simulator/src/main/java/com/hedera/block/simulator/BlockStreamSimulatorApp.java

```diff
@@ -60,9 +58,20 @@ public BlockStreamSimulatorApp(
 
         blockStreamConfig = configuration.getConfigData(BlockStreamConfig.class);
 
-        streamingMode = blockStreamConfig.streamingMode();
-        millisecondsPerBlock = blockStreamConfig.millisecondsPerBlock();
-        delayBetweenBlockItems = blockStreamConfig.delayBetweenBlockItems();
+        SimulatorMode simulatorMode = blockStreamConfig.simulatorMode();
+        switch (simulatorMode) {
+            case PUBLISHER:
+                simulatorModeHandler = new PublisherModeHandler(blockStreamConfig, publishStreamGrpcClient);
+                break;
+            case CONSUMER:
+                simulatorModeHandler = new ConsumerModeHandler(blockStreamConfig);
+                break;
+            case BOTH:
+                simulatorModeHandler = new CombinedModeHandler(blockStreamConfig);
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown SimulatorMode: " + simulatorMode);
+        }
```

Comment:
True, but also that would be the final state and no more modes will be added. Maybe we could add factory in the future, because is out of the scope here. 👍

Reply:
Yes for sure, my note is for the future, maybe an issue could be opened if you deem necessary in order to not forget that.